### PR TITLE
`shutil.rmtree`: Add `onexc` parameter; add parameter defaults

### PR DESCRIPTION
**Pull request by AlexWaygood**

### CODE REVIEW

File: stdlib/shutil.pyi [hunk 1]
```diff
@@ -88,21 +88,34 @@ else:
     ) -> _PathReturn: ...
 
 _OnErrorCallback: TypeAlias = Callable[[Callable[..., Any], Any, Any], object]
+_OnExcCallback: TypeAlias = Callable[[Callable[..., Any], Any, Exception], object]
```
Comment (hauntsaninja) [inline stdlib/shutil.pyi:91]: I think the second argument is `str`

Comment (hauntsaninja) [inline stdlib/shutil.pyi:91]:
```suggestion
_OnExcCallback: TypeAlias = Callable[[Callable[..., Any], str, Exception], object]
```

File: stdlib/shutil.pyi [hunk 2]
```diff
@@ -88,21 +88,34 @@ else:
     ) -> _PathReturn: ...
 
 _OnErrorCallback: TypeAlias = Callable[[Callable[..., Any], Any, Any], object]
```
Comment (AlexWaygood) [inline stdlib/shutil.pyi:90]:
```suggestion
_OnErrorCallback: TypeAlias = Callable[[Callable[..., Any], str, Any], object]
```

**Comment by AlexWaygood** [comment]
I guess it would be better to use overloads here, actually. You shouldn't be providing `onexc` _and_ `onerror` on 3.12+.

**Review by hauntsaninja — APPROVED** [review]

**Pull request merged**